ReactCommon/jsi/JSCRuntime.cpp

```diff
@@ -630,7 +630,9 @@ jsi::String JSCRuntime::createStringFromUtf8(
     size_t length) {
   std::string tmp(reinterpret_cast<const char*>(str), length);
   JSStringRef stringRef = JSStringCreateWithUTF8CString(tmp.c_str());
-  return createString(stringRef);
+  auto result = createString(stringRef);
```

Comment:
it's weird that it's necessary to create a string ref (copying the string in the process) in order to create a string (copying it again)

Reply:
Agreed. Also strange that one must release a JSStringRef. I would have thought it would be done with rtti 🤷‍♂

Reply:
Yeah looks like their process is str -> JSStringRef -> jsi::String, though I'd imagine their jsi::String is just a wrapper around the JSStringRef.

Reply:
if it was just a wrapper around a string ref then freeing the ref would break the strings
(also it’s RAII not RTTI ;)